app/src/main/java/edu/cascadia/mobas/campusguidebook/data/model/AppRepository.java

```diff
@@ -10,80 +10,106 @@
-    private ClubDao Clubdao;
-    private LiveData<List<ClubModel>> allClub;
-
+    private AppRepository(Application app) {
+        if (mAppRepository != null) return;
+        mApplication = app;
+        mAppDatabase = AppDatabase.getInstance(app);
+        mAppRepository = this;
+    }
 
-    // below line is the create a variable
-    // for dao and list for all Club.
-    private SustainabilityDao Sustainabilitydao;
-    private LiveData<List<SustainabilityModel>> allSustainability;
+    // returns the AppRepository instance, creating if necessary
+    public static AppRepository getInstance(Application app) {
+        if (mAppRepository == null) {
+            mAppRepository = new AppRepository(app);
+        }
+        return mAppRepository;
+    }
 
+    // returns the AppRepository instance if it exists, otherwise null
+    public static AppRepository getInstance() {
+        return mAppRepository;
+    }
 
+    // METHODS TO RETURN LIVEDATA FROM APPDATABASE
+    // returns a list of all events
+    public LiveData<List<Event>> getAllEvents() {
+        return mAppDatabase.EventDao().getAllEvents();
+    };
 
-    // creating a constructor for our variables
-    // and passing the variables to it.
-    public appRepository(Application application) {
-        appDatabase database = appDatabase.getInstance(application);
-        Eventdao = database.EventDao();
 
+    // returns a list of all users
+    public LiveData<List<User>> getAllUsers() {
+        return mAppDatabase.UserDao().getAllUsers();
     }
 
+
+    // returns a list of all clubs
+    public LiveData<List<Club>> getAllClubs() {
+        return mAppDatabase.ClubDao().getAllClubs();
+    };
+
+
+    // returns a list of all sustainability
+    public LiveData<List<Sustainability>> getAllSustainability() {
+        return mAppDatabase.SustainabilityDao().getAllSustainability();
+    };
+
+
+
+
+
     // creating a method to insert the data to our database.
-    public void insert(EventModel model) {
-        new InsertEventAsyncTask(Eventdao).execute(model);
+    public void insert(Event model) {
+        new InsertEventAsyncTask(mAppDatabase.EventDao()).execute(model);
     }
 
     // creating a method to update data in database.
-    public void update(EventModel model) {
-        new UpdateEventAsyncTask(Eventdao).execute(model);
+    public void update(Event model) {
+        new UpdateEventAsyncTask(mAppDatabase.EventDao()).execute(model);
     }
 
     // creating a method to delete the data in our database.
-    public void delete(EventModel model) {
-        new DeleteEventAsyncTask(Eventdao).execute(model);
+    public void delete(Event model) {
+        new DeleteEventAsyncTask(mAppDatabase.EventDao()).execute(model);
     }
 
     // we are creating a async task method to insert new Event.
-    private static class InsertEventAsyncTask extends AsyncTask<EventModel, Void, Void> {
+    private static class InsertEventAsyncTask extends AsyncTask<Event, Void, Void> {
```

Comment:
These AsyncTask methods are fine for now. We will look at some new tools on 2/9 to use Executors and callbacks instead